src/Compilers/CSharp/Portable/Binder/FixedStatementBinder.cs

```diff
@@ -42,14 +42,19 @@ override protected ImmutableArray<LocalSymbol> BuildLocals()
             return ImmutableArray<LocalSymbol>.Empty;
         }
 
-        internal override ImmutableArray<LocalSymbol> GetDeclaredLocalsForScope(CSharpSyntaxNode node)
+        internal override ImmutableArray<LocalSymbol> GetDeclaredLocalsForScope(CSharpSyntaxNode scopeDesignator)
         {
-            return this.Locals;
+            if (_syntax == scopeDesignator)
```

Comment:
No, we want to fail fast here. If we proceed, we can end-up with very hard to diagnose bugs.